src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -542,6 +711,9 @@ private void ThrowIfConnectionTerminated(string invocationId)
 
         private void AddInvocation(InvocationRequest irq)
         {
+            SafeAssert(_connectionLock.CurrentCount == 0, "We're not in the Connection Lock!");
```

Comment:
Should we have a helper method that wraps SafeAssert for checking the connection lock? This is copy/pasted 9 times.

Reply:
Yeah, I was seeing that too. I'll probably wrap it up in a helper.